File: Android-Demo/ZXingDemo/app/src/main/java/com/wx/zxingdemo/base/BaseActivity.java
package com.wx.zxingdemo.base;

import android.Manifest;
import android.app.Activity;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.Toast;

import com.tbruyelle.rxpermissions2.RxPermissions;
import com.trello.rxlifecycle2.components.support.RxAppCompatActivity;

import io.reactivex.functions.Consumer;

public abstract class BaseActivity extends RxAppCompatActivity {

    public abstract int getLayout();

    public abstract void setDataBinding(ViewDataBinding binding);

    public abstract void initData();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initPermission();
        setDataBinding(DataBindingUtil.setContentView(this,getLayout()));
        initData();
    }

    private void initPermission() {
        RxPermissions rxPermissions = new RxPermissions(this);
        rxPermissions.request(Manifest.permission.CAMERA,Manifest.permission.WRITE_EXTERNAL_STORAGE)
                .compose(this.<Boolean>bindToLifecycle())
                .subscribe(new Consumer<Boolean>() {
                    @Override
                    public void accept(Boolean aBoolean) throws Exception {
                        if (aBoolean) {

                        } else {
                            showToast("请授予权限否则无法测试");
                        }
                    }
                });


    }

    public void openActivity(Class<? extends Activity> activity) {
        Intent intent = new Intent(this,activity);
        startActivity(intent);
    }

    public void showToast(String data) {
        if (data != null && !data.isEmpty())
            Toast.makeText(this,data,Toast.LENGTH_SHORT).show();
    }

}

File: 19/Review/app/src/main/java/com/wx/review/base/BaseActivity.kt
package com.wx.review.base

import android.app.Activity
import android.content.Intent
import android.databinding.DataBindingUtil
import android.databinding.ViewDataBinding
import android.os.Bundle
import android.widget.Toast

abstract class BaseActivity : Activity(), BaseView{

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setDataBinding(DataBindingUtil.setContentView<ViewDataBinding>(this,layoutId))
        initData()
    }

    override fun openActivity(activity: Class<out Activity>) {
        startActivity(Intent(this,activity))
    }

    override fun showToast(data: String) {
        if (data === null) return

        Toast.makeText(this,data,Toast.LENGTH_SHORT).show()
    }

}